src/__tests__/cli.ts

```diff
@@ -1,32 +1,61 @@
 import fs from 'fs';
 import path from 'path';
 import util from 'util';
-import childProcess from 'child_process';
+import { CliArguments } from '..';
 
 const mkdir = util.promisify(fs.mkdir);
 const rmdir = util.promisify(fs.rmdir);
 const writeFile = util.promisify(fs.writeFile);
 const readFile = util.promisify(fs.readFile);
 
-function exec(
-  command: string,
-  options: { cwd: string },
-): Promise<{ exitCode: number | null; stdout: string; stderr?: string }> {
-  return new Promise((resolve) => {
-    try {
-      const buffer = childProcess.execSync(command, options);
-      resolve({
-        exitCode: 0,
-        stdout: buffer.toString(),
-      });
-    } catch (ex) {
-      resolve({
-        exitCode: ex.status,
-        stdout: ex.stdout.toString(),
-        stderr: ex.stderr.toString(),
-      });
-    }
-  });
+async function exec(
+  testProjectDir: string,
+  { init = false, flow = false, update = false }: Partial<CliArguments> = {},
+): Promise<{ exitCode: number | null; stdout: string; stderr: string }> {
+  const originalExit = process.exit;
+  const originalCwd = process.cwd();
+  const originalConsole = {
+    log: console.log,
+    warn: console.warn,
+    error: console.error,
+  };
+
+  try {
+    let exitCode: number | null = null;
+    let stdout = '';
+    let stderr = '';
+
+    const appendStdout = (...args: any[]): void => {
+      stdout += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    const appendStderr = (...args: any[]): void => {
+      stderr += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    console.log = appendStdout;
+    console.warn = appendStdout;
+    console.error = appendStderr;
+
+    process.exit = (code: number): never => {
+      exitCode = exitCode ?? code;
+      return undefined as never;
+    };
```

Comment:
replace `process.exit` to capture the code but not _actually_ kill the process.

**There is a caveat here** that the execution doesn't halt either.

We _could_ throw an Error here instead of just returning, but that will force it to into the `catch` block of `main` which produces additional `stderr` output and tries to change the exit code (I've got protection here to prevent the `exitCode` being overridden, but that was for the `update` cases) which would cause some of the scenarios to fail as they are currently written (they check that there is no `stderr` output).